### ISSUE TITLE
Major lag issue associated with greenhouse client

### ISSUE THREAD

**Issue by zts16** [ForestryMC/ForestryMC#2092]
I'm trying to use Forestry version 5.8.0.264 for Minecraft 1.12.2 and have been experiencing major lag to the point of it being unplayable. I've never experienced lag on this computer before. I noticed that this message is constantly being returned in the output log:

[Forestry Greenhouse/WARN] [forestry]: GREENHOUSE TAKING xxxx ms LONGER THAN NORMAL

I removed Forestry and the lag issue appeared to be solved. Other mods that I am using (not including necessary core mods):
Biomes o' Plenty
Familiar Fauna
Galacticraft
Immersive Engineering/Petroleum/Railroading
Mekanism
NEI/JEI
Redstone Flux/Thermal Foundation/Expansion/Dynamics
Serene Seasons
Tough As Nails

Thanks.

**Comment by zts16** [comment]
Update: when running the game with only Forestry installed, this issue does not occur. I will try adding mods one at a time and see if I can figure out what's going on.

**Comment by Darianna77** [comment]
Have you found out wich mod is it? Because i have the same problems in the pack that i play :(

**Comment by temp1011** [comment]
Best way to check is to remove half of the mods and try to reproduce. Then you can check half of your mods at once. Repeat with the 'bad' half until you're left with one mod.

**Comment by Darianna77** [comment]
Its not so easy in a big modpack with over 100 mods :)

**Comment by temp1011** [comment]
I agree, but it should only take 7 tries max. Some stuff can probably be taken out without much worry like JEI and CoFH stuff. Some stuff is probably more likely to be the problem like anything with a coremod.

**Comment by temp1011** [comment]
Or just start with the mods that OP mentioned. Then there aren't too many at all.

**Comment by zts16** [comment]
It seems like the conflict was caused by one of the CoFH mods. Removing those seemed to fix the problem, and I decided that I could live without them now that I have Immersive Engineering. The game worked fine for a while, but then an error with my computer's RAM caused all games to stop working on my computer. I don't know if the bug had a role in causing this problem, or vice-versa. After resetting my computer's NVRAM, the game resumed working better than ever before. I am now running all of the mods that I listed above except for the CoFH mods, Immersive Railroading, and Galacticraft. The latter two were removed due to personal preference, not because they weren't working.

**Comment by Aevaris** [comment]
TL;DR: This is the first sign of general performance issues and especially RAM issues. I don't think anything is wrong with Forestry and my problem described below ultimately ended up being that JourneyMap does not play well with shaders.

Not sure if this is helpful or not (sorry if not, but intent is to be). I'm playing with ~124 mods on 1.12.2, including forestry at newest and cofh at newest and forge at newest. I'm playing with 8gigs allocated on 8700k and 1080ti and NVMe M.2 drive. So my computer should be able to handle a fair bit.

I briefly experience this problem also, but I had changed multiple things (added shaders, upgraded forge due to upgrading COFH core as newest COFH core requires newest forge [as of time of writing]) so I've been toying with it for a bit and can add some of my findings.

Things started as trying to improve my shader performance. I read somewhere MC does better with less memory allocated, so I dropped from 10gigs to 4gigs max. My performance was much worse, so I dropped shaders out completely, downgraded Forge and down graded COFH core. Still this issue. So I raised my allocated RAM to 8gigs (no shaders), problem went away. Brought COFH core and forge back in, no problem. Brought Shader back in and kicked out JourneyMap and everything has been fine since.
